models/repo.go

```diff
@@ -470,6 +470,12 @@ func MigrateRepository(u *User, name, desc string, private, mirror bool, url str
 		return repo, fmt.Errorf("create update hook: %v", err)
 	}
 
+	// Check if repository is empty.
+	_, _, err = com.ExecCmdDir(repoPath, "git", "rev-parse", "--verify", "--quiet", "HEAD")
```

Comment:
stderr is silenced by `--quiet`. I running `git log` on a repository with thousands of commits might not be a good idea because it might take a long time. This approach only finds the commit that is referenced by `HEAD`, the exit status will be 0 (a commit found) or non-zero (no commit found).